orderbook/src/fee.rs

```diff
@@ -263,8 +266,7 @@ impl MinFeeCalculating for MinFeeCalculator {
         Ok((min_fee, official_valid_until))
     }
 
-    // Returns true if the fee satisfies a previous not yet expired estimate, or the fee is high enough given the current estimate.
-    async fn is_valid_fee(&self, sell_token: H160, fee: U256, app_data: AppId) -> bool {
+    async fn validate_fee(&self, sell_token: H160, fee: U256, app_data: AppId) -> Result<U256, ()> {
```

Comment:
Interesting use of `Result` for invalid fees. I like it!

One idea would be to define a `struct InsufficientFeeError(U256)` so that:
- The caller knows the minimum in case its too small.
- Allows `impl From<InsufficientFeeError> for MyHighLevelError` conversion trait implementations for better `?` support. Otherwise, we would need `.map_err(|_| MyHighLevelError::InsufficienFeeError)?` in the calling context.

Not sure if my ideas are a bit overkill, just wanted to put them out there. Either way, I like the use of `Result` here over the bool.